### PR TITLE
Implement Voucher Form Service Unit Tests

### PR DESCRIPTION
**Pull request by jniles**
This PR implements tests for the VoucherFormService, since it is used in several important clientside modules.  It also reinforces the need for more unit tests as mentioned in #337.

In order to accomplish these tests in a light-weight manner, I proposed and implemented #1684.  The enhancement decouples `bhima.services` from the main `bhima` module, so that we can inject only the constants without having to import all the rest of the application dependencies.

----
Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

File: client/src/js/constants/bhConstants.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+angular.module('bhima.constants')
```
Comment (sfount) [inline client/src/js/constants/bhConstants.js:1]: :+1:

File: client/src/modules/vouchers/voucher-form.service.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ angular.module('bhima.services')
 
 VoucherFormService.$inject = [
   'VoucherService', 'bhConstants', 'SessionService', 'VoucherItemService',
-  'CashboxService', 'AppCache', 'Store', 'AccountService', '$timeout', '$translate'
+  'CashboxService', 'AppCache', 'Store', 'AccountService', '$timeout', '$translate',
```
Comment (sfount) [inline client/src/modules/vouchers/voucher-form.service.js:6]: I don't think we need a comma here.

File: test/client-unit/services/VoucherForm.spec.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    expect(form.hasCacheAvailable()).to.equal(false);
+  });
+
+  it('#onChanges() calls #validate()', () => {
+    form.validate = chai.spy();
+    form.onChanges();
+    expect(form.validate).to.have.been.called.exactly(1);
+  });
+
+  it('#validate() calculate totals', () => {
```
Comment (sfount) [inline test/client-unit/services/VoucherForm.spec.js:160]: These tests are awesome :+1:. If we can cover our financial components like this the application will be much stronger. Great work!

**Review by sfount — APPROVED** [review]
LGTM!

File: client/src/modules/vouchers/voucher-form.service.js (hunk 2 above)
Comment (jniles) [reply client/src/modules/vouchers/voucher-form.service.js:6]: Callin' out the guy who doesn't have eslint installed. :D

**Pull request merged**